8354466: Open some misc Swing bugs 9

alisenchung opened the pull request:
Updating and opening some misc swing bugs

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8354466](https://bugs.openjdk.org/browse/JDK-8354466): Open some misc Swing bugs 9 (**Bug** - P4)


### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**) Review applies to [4875b8ff](https://git.openjdk.org/jdk/pull/24703/files/4875b8ff12669dc226d9cf59301651e4209c8998)
 * [Alexander Zuev](https://openjdk.org/census#kizune) (@azuev-java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/24703/head:pull/24703` \
`$ git checkout pull/24703`

Update a local copy of the PR: \
`$ git checkout pull/24703` \
`$ git pull https://git.openjdk.org/jdk.git pull/24703/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 24703`

View PR using the GUI difftool: \
`$ git pr show -t 24703`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/24703.diff">https://git.openjdk.org/jdk/pull/24703.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/24703#issuecomment-2811086204)
</details>

prsadhuk commented on test/jdk/javax/swing/JSplitPane/bug4820080.java:
```diff
@@ -0,0 +1,87 @@
+ * @test
+ * @bug 4820080 7175397
+ * @summary RFE: Cannot Change the JSplitPane Divider Color while dragging
+ * @library /open/test/jdk/java/awt/regtesthelpers
```
library path needs to be corrected

kumarabhi006 commented on test/jdk/javax/swing/JSplitPane/bug4820080.java:
```diff
@@ -0,0 +1,87 @@
+        frame.add(box);
+
+        JPanel jleft = new JPanel();
+        jleft.setBackground(Color.darkGray);
```
You may change from lowercase to uppercase for constants here and below
```suggestion
        jleft.setBackground(Color.DARK_GRAY);
```

honkar-jdk commented on test/jdk/javax/swing/JPasswordField/bug4382819.java:
```diff
@@ -0,0 +1,88 @@
+            Robot robot = new Robot();
+            robot.setAutoDelay(250);
+            SwingUtilities.invokeAndWait(() -> createTestUI());
+            robot.waitForIdle();
```
Add some delay after testUI ?

```suggestion
            robot.waitForIdle();
            robot.delay(1000);
```

honkar-jdk commented on test/jdk/javax/swing/JPasswordField/bug4382819.java:
```diff
@@ -0,0 +1,88 @@
+
+            SwingUtilities.invokeAndWait(() -> passwordField.setEnabled(false));
+            robot.waitForIdle();
+            robot.delay(1000);
```
delay can be reduced here

```suggestion
            robot.delay(250);
```

honkar-jdk replied:
delay was removed instead of reducing it here. Please add back the delay.

alisenchung replied:
delay is present via setAutoDelay(250)

honkar-jdk commented on test/jdk/javax/swing/JSplitPane/bug4820080.java:
```diff
@@ -0,0 +1,87 @@
+        Box box = new Box(BoxLayout.Y_AXIS);
+        frame.add(box);
+
+        JPanel jleft = new JPanel();
+        jleft.setBackground(Color.DARK_GRAY);
+        JPanel jright = new JPanel();
+        jright.setBackground(Color.DARK_GRAY);
```
using setPreferredSize() makes the UI look better and the divider is not skewed to the left
Same for left and right panels below.

```suggestion
        JPanel jleft = new JPanel();
        jleft.setBackground(Color.DARK_GRAY);
        jleft.setPreferredSize(new Dimension(100, 100));
        JPanel jright = new JPanel();
        jright.setBackground(Color.DARK_GRAY);
```

honkar-jdk reviewed (approved):
LGTM apart for minor inline suggestions

honkar-jdk commented on test/jdk/javax/swing/JSplitPane/bug4820080.java:
```diff
@@ -0,0 +1,91 @@
+        box.add(new JSeparator());
+        box.add(Box.createVerticalStrut(5));
+
+        Panel left = new Panel();
+        left.setBackground(Color.DARK_GRAY);
+        Panel right = new Panel();
+        right.setBackground(Color.DARK_GRAY);
```
You missed adding setPreferredSize() to the left and right panels

honkar-jdk reviewed (commented):
You missed adding the preferred size to the correct panel vars. Please check inline comments

alisenchung commented:
just updated inlines, also reverted other test to manual test because grabbing pixels of JPasswordField is inconsistent across platforms and causes issues with automating the test

DamonGuy commented on test/jdk/javax/swing/JPasswordField/bug4382819.java:
```diff
@@ -0,0 +1,86 @@
+ * @summary Tests the correctness of color used for the disabled passwordField.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main bug4382819
```
```suggestion
 * @run main/manual bug4382819
```
Looks like this test is manual when testing. Add the manual tag to the header.

alisenchung commented:
/integrate
